### PR TITLE
llmservice reconciler implementation

### PR DESCRIPTION
**Pull request by kfswain**
Adding the LLMServiceReconciler and updating the datastore. Consuming the LLMService is considered out of scope of this PR

### CODE REVIEW

File: pkg/ext-proc/backend/llmservice_reconciler.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+func (c *LLMServiceReconciler) updateDatastore(service *v1alpha1.LLMService) {
+
+	for _, ref := range service.Spec.PoolRef {
+		if strings.Contains(strings.ToLower(ref.Kind), strings.ToLower("LLMServerPool")) && ref.Name == c.ServerPoolName {
+			c.Datastore.LLMServices.Store(service.Name, service)
+			klog.Info("Stored!!")
+			return
+		}
+	}
+	klog.Info("Deleted :(")
```
Comment (ahg-g) [inline pkg/ext-proc/backend/llmservice_reconciler.go:55]: more descriptive message pls

File: pkg/ext-proc/backend/llmservice_reconciler.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+}
+
+func (c *LLMServiceReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&v1alpha1.LLMService{}).
+		Complete(c)
+}
+
+func (c *LLMServiceReconciler) updateDatastore(service *v1alpha1.LLMService) {
+
```
Comment (ahg-g) [inline pkg/ext-proc/backend/llmservice_reconciler.go:47]: nit, remove empty line

File: pkg/ext-proc/backend/llmservice_reconciler.go [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+	for _, ref := range service.Spec.PoolRef {
+		if strings.Contains(strings.ToLower(ref.Kind), strings.ToLower("LLMServerPool")) && ref.Name == c.ServerPoolName {
+			c.Datastore.LLMServices.Store(service.Name, service)
+			klog.Info("Stored!!")
```
Comment (ahg-g) [inline pkg/ext-proc/backend/llmservice_reconciler.go:51]: make it more descriptive

**Review by ahg-g — COMMENTED** [review]
There is a follow up PR to do the model name re-write, right?

**Comment by ahg-g** [comment]
> There is a follow up PR to do the model name re-write, right?

This is already mentioned in description :)

**Comment by kfswain** [comment]
A couple small linter-error fixes made here as well.

**Pull request merged**